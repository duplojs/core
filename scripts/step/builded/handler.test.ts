import { readFile, writeFile } from "fs/promises";
import { resolve } from "path";
import { BuildedHandlerStep } from "./handler";
import { HandlerStep } from "../handler";
import { Response } from "@scripts/response";

it("BuildedHandlerStep", async() => {
	const handlerFunction = () => new Response(300, "test", 11);

	const step = new HandlerStep(handlerFunction, ["test"]);
	const buildedCutStep = new BuildedHandlerStep(step);

	expect(buildedCutStep.step).toBe(step);

	expect(buildedCutStep.handlerFunction).toBe(handlerFunction);

	expect(buildedCutStep.toString(4)).toBe(
		await readFile(resolve(import.meta.dirname, "__data__/handler.txt"), "utf-8"),
	);
});
