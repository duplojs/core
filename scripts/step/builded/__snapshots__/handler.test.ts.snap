// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`BuildedHandlerStep 1`] = `
"
		
/* beforeHandlerStep(index: 4) */
/* end_block */


		result = this.steps[4].handlerFunction(floor.pickup, request);

		
/* beforeTreatResultHandlerStep(index: 4) */
/* end_block */


		if(result instanceof this.Response){

				let temp = this.steps[4].responseZodSchema.safeParse(result);

				if(!temp.success){
					throw new this.ContractResponseError(temp.error, result);
				}
			
break Execution;
}

		
/* afterHandlerStep(index: 4) */
/* end_block */

		"
`;
